### DISCUSSION TITLE
Package built using internal package is not bundling the internal package's type-declarations

### DISCUSSION THREAD

**Discussion by rajeshdavidbabu** [vercel/turborepo#2667] [Help] [answered]
Hi Team,

  We currently have a mono-repo setup with three packages:
  - App
  - Web
  - Common (shared internal package)
  
  The project is built on react and typescript. Based on the bundling example here https://turbo.build/repo/docs/handbook/publishing-packages/bundling, 
  
  Considering the above scenario, when I build let say Web package, its built type declarations are pointing to Common. 
  Eg: import { ButtonProps, ListProps } from Common. 
  
  But I would rather expect all the type-declarations of the Common package to be included in the Web, without references to Common, because Common will not published. 
  
  Tool used to build: tsup

**Comment by EarlGeorge** [comment]
didn't get it. but if you are using internal packages you don't need tsup at all, otherwise if bundling external package with type declaration you need to pin it in package.json if you don't prefer **cjs** format change it to **esm**

**Reply by rajeshdavidbabu** [reply, marked as answer]
My problem is similar to this and I solved it via this solution https://dev.to/egoist/rollup-dts-file-using-tsup-2579?signin=true

**Reply by EarlGeorge** [reply]
congrats
